src/utils/aquarius.ts

```diff
@@ -126,16 +126,19 @@ export async function getAssetsNames(
 export async function transformDDOToAssetSelection(
   datasetProviderEndpoint: string,
   ddoList: DDO[],
-  selectedAlgorithms?: PublisherTrustedAlgorithm[]
+  selectedAlgorithms?: PublisherTrustedAlgorithm[],
+  ddoChainId?: number
```

Comment:
just checked, all works in the end but feels super ineffective as I assume we are first getting all algos of all networks and then filter them? So a big impact optimization would be to rather modify the Elasticsearch query to begin with which should be way faster in the end and we can ditch the js filtering. But just first thoughts, maybe I'm missing something why we do it like this

Reply:
Yep you are right, I have thought about this also, I'll try to figure out a way of getting just the algos instead of getting them from all networks and filter them